executor/executor_test.go

```diff
@@ -7562,3 +7562,9 @@ func (s *testSuite) TestIssue22231(c *C) {
 	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1292 Truncated incorrect datetime value: '2020-05-28 23:59:59 00:00:00'"))
 	tk.MustExec("drop table if exists t_issue_22231")
 }
+
+func (s *testSuite) TestIssue22201(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustQuery("SELECT HEX(WEIGHT_STRING('ab' AS BINARY(1000000000000000000)));").Check(testkit.Rows("<nil>"))
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1301 Result of cast_as_binary() was larger than max_allowed_packet (67108864) - truncated"))
```

Comment:
MySQL:
```
Result of weight_string() was larger than max_allowed_packet (67108864) - truncated
```

Reply:
In MySQL8, it is 
`Result of cast_as_binary() was larger than max_allowed_packet (67108864) - truncated`

Reply:
Please check this :
```
mysql> SELECT HEX(WEIGHT_STRING('ab' AS char(1000000000000000000)));
+-------------------------------------------------------+
| HEX(WEIGHT_STRING('ab' AS char(1000000000000000000))) |
+-------------------------------------------------------+
| NULL                                                  |
+-------------------------------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                             |
+---------+------+-------------------------------------------------------------------------------------+
| Warning | 1301 | Result of weight_string() was larger than max_allowed_packet (67108864) - truncated |
+---------+------+-------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

Reply:
I use MySQL  8.0.21, and the warning is `Result of cast_as_binary() was larger than max_allowed_packet (67108864) - truncated`.

Reply:
The sql of this issue is `"SELECT HEX(WEIGHT_STRING('ab' AS BINARY(1000000000000000000)))`.

Reply:
`SELECT HEX(WEIGHT_STRING('ab' AS char(1000000000000000000)));` also casues `makeslice: cap out of range`.

Reply:
OK, I have fixed it, PTAL.